web-platform/functions/src/index.ts

```diff
@@ -418,3 +421,111 @@ export const offboard = functions.https.onRequest(async (request, response) => {
 
   response.status(301).redirect(OFFBOARD_URL);
 });
+
+/**
+ * Waitlist function, to collect user info for Sendgrid.
+ */
+export const waitlist = functions
+  .runWith({ secrets: ["SENDGRID_API_KEY"] })
+  .https.onRequest(async (request, response) => {
+    useCors(request, response, async () => {
+
+      if (request.method !== "POST") {
+        response.status(500).send("Only POST and OPTIONS methods are allowed.");
+        return;
+      }
+
+      if (!(request.headers && request.headers["content-type"]?.startsWith("multipart/form-data"))) {
+        response.status(500).send("Only Content-type: multipart/form-data is allowed.");
+        return;
+      }
+
+      const busboy = Busboy({ headers: request.headers });
+      let formData = new Map();
+
+      busboy.on("field", (fieldname: string, value: string) => {
+        formData.set(fieldname, value);
+      });
+
+      // In production. GCP cloud function receive the raw body as a
+      // Buffer. Unit tests don't have a way to set this independently,
+      // so fall back to looking for the form in the request.body if rawBody is missing.
+      let rawForm;
+      if (request.rawBody) {
+        rawForm = request.rawBody;
+      } else {
+        rawForm = (new TextEncoder()).encode(request.body);
+      }
```

Comment:
We can make this const:
const rawForm = request.rawBody || (new TextEncoder()).encode(request.body)